pkg/controller/util/helpers.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"os"
+	"strings"
+)
+
+// IsValidAppNamespace returns an error when the namespace passed is not present in the APP_NAMESPACES environment variable provided to the operator.
+func IsValidAppNamespace(crNamespace string, operatorNamespace string, name string) bool {
+	appNamespacesEnvVar, found := os.LookupEnv("APP_NAMESPACES")
+	if !found {
+		return false
+	}
+
+	if operatorNamespace == crNamespace {
```

Comment:
I would put this above the check for "APP_NAMESPACES", since if that's not set, CRs in the operator namespace should still be considered valid.